enhancements/update/versioning-cincinnati-api.md

```diff
@@ -0,0 +1,116 @@
+3. Move to a generic un-versioned URI for cincinnati
+4. Cincinnati continues to support older clients after adding support for new clients for foreseeable future.
+5. Add metrics to identify hits on URIs and versions.
+
+### Non-Goals
+
+## Proposal
+
+### Implementation
+
+#### Version information in json schema 
+Embed the version information in json schema. The updated json schema should look like:
+```json
+{
+    "version": 1,
+    "nodes": [],
+    "edges": []
+}
+```
+
+#### Media Type Versioning 
+Currently, the versioning is done by using the versioned URI and does not support media type versioning. The user should 
+be able to request a specific version of cincinnati graph by supplying the version in media type field of the request. 
+for e.g. `ContentType: application/vnd.redhat.cincinnati.v1+json`. Here are some of the 
+[examples of media type versioning](https://github.com/opencontainers/image-spec/blob/v1.0.1/media-types.md).       
+In the event where the media type is not provided, or `application/json` is provided, cincinnati should return a default 
+response that is compatible with earlier versions (v1) so as not to break old clusters from upgrading. 
+
+
+#### URI Change
+The current URI follows `{URL}/v1/graph`. The new un-versioned URI will follow `{URL}/graph`. The existing URI can be 
+redirected to the new URI. This can be done through cincinnati. In the case where an old URI is being called, cincinnati 
+will by default return the graph that is compatible with `v1`.      
+
+
+#### Version metrics
+Cincinnati should also implement a way to understand how many clients are consuming the various versions. 
+This can be done by implementing metrics for old URI vs new URI. Similarly, metrics should be implemented to understand
+the consumption of various media types (versions). This will help us to make decisions on deprecating the old versions 
+and URI. 
+
+
+### Risks and Mitigations
+* As there are some customers that have old URI hardcoded while creating the cluster, we will have to support old URI 
+  for indefinite period of time. We can redirect the old URI to new one to avoid maintaining different URIs, but we'll 
+  still have to maintain different versions and v1 for indefinite period of time. 
+
+
+## Design Details
+
+### Test Plan
+* cincinnati tests to make sure we are not breaking older versions 
+* cincinnati returns correct resource version when the content type is provided.
+
+### Graduation Criteria
+GA. When it works, we ship it.
+
+
+### Version Skew Strategy
+No special consideration.
+
+## Implementation History
+
+## Drawbacks
+
+
+## Alternatives
+We can continue with the current URI based versioning for the API and adding version information in the json schema.     
+**Drawback**: For every minor non-breaking change with a bumped version, the client will have to change the URI with 
+the new version to get upgrades.
```

Comment:
Main issue with this is that [proactive negotiation][1] makes compat between simple clients and a smarter server easy, and that's the approach we're aiming for with the Cincinnati service and the cluster-version operator.  Teaching the CVO to walk possible URIs or hit a list-of-available-versions endpoint would be more complicated.

Another alternative is adding content-type negotiation, but sticking with the existing URI.  You should add a subsection on that, and point out that we're avoiding it because folks might find the dissonance confusing while reading docs talking about v2 content from a v1 URI.

[1]: https://datatracker.ietf.org/doc/html/rfc7231#section-3.4.1